Sources/KlaviyoSwift/StateManagement/APIRequestErrorHandling.swift

```diff
@@ -91,13 +92,20 @@ func handleRequestError(
         environment.emitDeveloperWarning("A data encoding error occurred during transmission.")
         return .deQueueCompletedResults(request)
 
+    case .dataDecodingError:
+        environment.emitDeveloperWarning("A data encoding error occurred during transmission.")
+        return .deQueueCompletedResults(request)
+
     case .invalidData:
         environment.emitDeveloperWarning("Invalid data supplied for request. Skipping.")
         return .deQueueCompletedResults(request)
 
     case let .rateLimitError(retryAfter):
+        guard let retryInfo else { return .deQueueCompletedResults(request) }
```

Comment:
It looks like we only care about the `retryInfo` when the error type is `.networkError` or `.rateLimitError`. For all other error types, we ignore the `retryInfo`. So if I were to unwrap the `retryInfo` at the beginning of the function's execution, I'd be unwrapping a value that's not needed for the majority of the error cases. If the `retryInfo` were `nil`, I'd also be unnecessarily returning early for error cases that don't require the `retryInfo`. 

The reason I made `retryInfo` optional is because this PR introduces a new situation where we don't want to retry the API request. If we successfully fetch the forms data from the fullforms endpoint, but we're unable to successfully decode the data, we probably don't want to retry the network request because the server will just give us the bad data again. Therefore I want to call `handleRequestError`, but I want to set `nil` for the `retryInfo` parameter. See [here](https://github.com/klaviyo/klaviyo-swift-sdk/pull/240/files#diff-c53c42711317995bbc065d8e1e4842a9108085cdb54786e4874c334508ba5c71R373).

Reply:
> If the retryInfo were nil, I'd also be unnecessarily returning early for error cases that don't require the retryInfo.

totally makes sense 

> The reason I made retryInfo optional is because this PR introduces a new situation where we don't want to retry the API request.

We may have a future where we might want to handle retries when the full forms apis starts enforcing it. But as you mentioned for now we can get away w/o it.